File: src/main/java/com/wanderers/hotelier_webservices/server/dao/api/CustomerDao.java
package com.wanderers.hotelier_webservices.server.dao.api;

import com.wanderers.hotelier_webservices.server.exception.CustomerDaoException;

/**
 * Data access contract for the customer entity
 */
public interface CustomerDao {

    /**
     * Check if the customer exists in the datasource
     *
     * @param customerId
     * @return
     * @throws CustomerDaoException
     */
    Boolean isExistingCustomer(int customerId) throws CustomerDaoException;
}

File: src/main/kotlin/com/wanderers/hotelier_webservices/server/dao/api/CustomerDao.kt
package com.wanderers.hotelier_webservices.server.dao.api

import com.wanderers.hotelier_webservices.server.exception.CustomerDaoException

/**
 * Data access contract for the customer entity
 */
interface CustomerDao {

    /**
     * Check if the customer exists in the datasource
     *
     * @param customerId
     * @return
     * @throws CustomerDaoException
     */
    @kotlin.Throws(CustomerDaoException::class)
    fun isExistingCustomer(customerId: Int): Boolean
}